Source/Core/Core/IOS/Network/IP/Top.h

```diff
@@ -74,6 +76,29 @@ class NetIPTopDevice : public Device
   void Update() override;
 
 private:
+  // MSVC std::future default constructor workaround
+  struct DefIPCReply : public IPCReply
+  {
+    DefIPCReply() : IPCReply(0) {}
+    DefIPCReply(IPCReply reply) : IPCReply(reply) {}
+  };
+
+  struct async_task
+  {
+    IOS::HLE::Request request;
+    std::packaged_task<std::future<DefIPCReply>()> task;
+    std::future<DefIPCReply> future;
+  };
+
+  template <typename Method, typename Request>
+  std::optional<IPCReply> AsyncTask(Method method, const Request& request)
+  {
+    std::packaged_task<std::future<DefIPCReply>()> task(
+        [this, method, request] { return std::async(std::launch::async, method, this, request); });
+    m_tasks.emplace_back(async_task{request, std::move(task)});
+    return std::nullopt;
```

Comment:
Is this supposed to only ever return std::optional?

Reply:
It used to return `GetNoReply()` which is now represented as a [std::optional<IPCReply>](https://github.com/dolphin-emu/dolphin/blob/d19d28e03088cddf9b7b8a1c143564d4f7185e48/Source/Core/Core/IOS/Network/IP/Top.cpp#L334) with the value `std::nullopt`.